File: src/test/java/kitchenpos/menus/tobe/domain/DisplayedNameTest.java
package kitchenpos.menus.tobe.domain;


import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.NullAndEmptySource;

import static org.assertj.core.api.Assertions.assertThatThrownBy;

class DisplayedNameTest {
    @DisplayName("메뉴그룹의 이름은 비어있을 수 없다.")
    @NullAndEmptySource
    @ParameterizedTest
    void create(final String name) {
        assertThatThrownBy(() -> new DisplayedName(name))
                .isInstanceOf(IllegalArgumentException.class);
    }
}

File: src/test/java/kitchenpos/menus/tobe/domain/DisplayedNameTest.kt
package kitchenpos.menus.tobe.domain

import io.kotest.assertions.throwables.shouldThrow
import io.kotest.core.spec.style.ExpectSpec
import io.kotest.matchers.shouldBe
import kitchenpos.menus.tobe.infra.FakePurgomalumClient

private val profanityWordValidator = ProfanityWordValidator(FakePurgomalumClient("후라이드", "양념"))

class DisplayedNameTest : ExpectSpec({
    expect("금지어가 포함된 이름을 생성하면 예외가 발생한다") {
        val target = "후라이드 양념치킨"

        shouldThrow<IllegalArgumentException> {
            MenuName(target, profanityWordValidator)
        }
    }

    expect("금지어가 포함되지 않은 이름을 생성하면 이름이 생성된다") {
        val target = "간장치킨"

        val actual = MenuName(target, profanityWordValidator)

        actual.value shouldBe target
    }
})
